VideoApi/VideoApi.Services/AzureStorageServiceBase.cs

```diff
@@ -106,5 +138,32 @@ private async Task<string> GenerateSasToken(BlobSasBuilder builder, bool useUser
 
             return blobSasQueryParameters.ToString();
         }
+
+        public async Task<bool> ReconcileFilesInStorage(string fileNamePrefix, int count)
+        {
+            var allBlobs = await GetAllBlobNamesByFilePathPrefix(fileNamePrefix);
+            var emptyBlobs = await GetAllEmptyBlobsByFilePathPrefix(fileNamePrefix);
+
+            if (allBlobs.Count() < count || !allBlobs.Any())
+            {
+                var msg = $"ReconcileFilesInStorage - File name prefix :" + fileNamePrefix + "  Expected: " + count + " Actual:" + allBlobs.Count().ToString();
+                throw new AudioPlatformFileNotFoundException(msg, HttpStatusCode.NotFound);
+            }
```

Comment:
Also any reason to only check if the count is lower and not equal? Is it okay for the count to be higher?

Reply:
there is no chance of zero coming in as parater from scheduler job, if this is called it will definitely have atleast 1 as count

Reply:
I have now added validaiton to request to check if file count is zero or negative.

Reply:
I'd still be inclined to check in here as we want it to make sense as a unit rather than relying on checks outside. Also how about if the count is greater than expected?

Reply:
as per the requirement it is ok to get more number of files than expected, so in this situation it should not error.